src/rest.js

```diff
@@ -0,0 +1,83 @@
+const debug = require('debug')('prismarine-realms')
+const fetch = require('node-fetch')
+
+const { PlatformConstants, BedrockRealmsRelyingParty } = require('./constants')
+const { formatBedrockAuth, formatJavaAuth } = require('./util')
+
+module.exports = class Rest {
+  constructor (authflow, platform) {
+    this.authflow = authflow
+    this.platform = platform
+  }
+
+  get (route, options) {
+    return this.prepareRequest({ ...options, route, method: 'get' })
+  }
+
+  post (route, options) {
+    return this.prepareRequest({ ...options, route, method: 'post' })
+  }
+
+  put (route, options) {
+    return this.prepareRequest({ ...options, route, method: 'put' })
+  }
+
+  delete (route, options) {
+    return this.prepareRequest({ ...options, route, method: 'delete' })
+  }
+
+  async prepareRequest (request) {
+    const { Host, UserAgent } = PlatformConstants(this.platform)
+
+    const url = `${Host}${request.route}`
+
+    const headers = {
+      'Client-Version': '0.0.0',
+      'User-Agent': UserAgent
+    }
+
+    if (request.auth !== false) {
```

Comment:
this isn't good, we end up doing these checks every API call. This should be one once on construction, you can use a static initializer method if you need to do something on async that otherwise can't be done in the default constructor

Reply:
This was leftover from when I was testing backup downloads, it errored with the auth header. I'll remove this